File: api/dto/website.go
package dto

type Content struct {
	Type     string    `json:"type" validate:"required,oneof=text image list container"`
	ID       string    `json:"id,omitempty"`
	Value    string    `json:"value"`
	Path     string    `json:"path,omitempty"`
	Elements []Content `json:"elements,omitempty" validate:"dive,content"`
}

type UpdateWebsiteDto struct {
	Name      string    `json:"name" validate:"required,min=3"`
	Content   []Content `json:"content" validate:"required,dive,content"`
	Redactors []string  `json:"redactors" validate:"required,min=1,dive,required"`
}

type CreateWebsiteDto struct {
	Name      string    `json:"name" validate:"required,min=3"`
	Content   []Content `json:"content" validate:"required,dive,content"`
	Redactors []string  `json:"redactors" validate:"required,min=1,dive,required"`
}

type GetWebsitesDto struct {
	ID   string `json:"id,omitempty"`
	Name string `json:"name"`
}

File: admin-dash/src/types/website.ts
export interface Content {
  type: 'text' | 'image' | 'container' | 'list'
  id?: string
  value: string
  path?: string
  elements?: Content[]
}

export interface Website {
  id: string
  name: string
  available_languages: string[]
}

export interface WebsiteDetail {
  id: string
  name: string
  language: string
  content: Content[]
}

export interface UpdateWebsiteDto {
  name: string
  language: string
  content: Content[]
  redactors: string[]
}

export interface CreateWebsiteDto {
  name: string
  language: string
  content: Content[]
  redactors: string[]
}

export const SUPPORTED_LANGUAGES = [
  { code: 'en', name: 'English' },
  { code: 'pl', name: 'Polish' },
  { code: 'de', name: 'German' },
  { code: 'fr', name: 'French' },
  { code: 'es', name: 'Spanish' },
] as const

export type LanguageCode = typeof SUPPORTED_LANGUAGES[number]['code']

